File: romanToInt.cpp
#include<iostream>
using namespace std;

class Solution {
public:
    int romanToInt(string s) {
        string roman[] = {"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};
        int nums[] =     {1000,900, 500, 400,100,  90, 50,  40, 10,   9,  5,   4,  1};
        int i=0;
        int out=0;
        string str = s;
        
        //int round=0;
        
        while(s.size() > 0)
        {
            //cout<<s<<endl;
            //cout<<nums[i]<<endl;
            //round++;
            //if(round>20)
             //   exit(0);
            if(s.find(roman[i])==0)
            {
                s=s.substr(roman[i].size());
                out+=nums[i];
            }
            else
            {
                i++;
            }
        }
        return out;
    }
    
};

int main()
{
    Solution sl;
    string  s = "LVIII";

    int out = sl.romanToInt(s);
    cout<<out<<endl;
    return 0;
}

File: romanToInt.py
class Solution:
    def romanToInt(self, s: str) -> int:
        nums = (('I', 1), ('IV', 4), ('V', 5), ('IX', 9), ('X', 10),
                ('XL', 40), ('L', 50), ('XC', 90), ('C', 100), ('CD', 400),
                ('D', 500), ('CM', 900), ('M', 1000))
        out = 0

        for i in range(len(nums) - 1, -1, -1):
            try:
                while s.index(nums[i][0]) == 0:
                    s = s[len(nums[i][0]):]
                    out += nums[i][1]
            except Exception:
                pass

        return out


sl = Solution()
s = "MCMXCIV"
print(sl.romanToInt(s))
